### PR TITLE
Reject xml accepts

### PR DESCRIPTION
**Pull request by eltmon**

### CODE REVIEW

**Comment by williewheeler** [comment]
The XML thing is a global issue, rather than something that attaches to a specific controller. So whatever solution we put in place should deal with it in that way, as opposed to ad hoc changes, if possible. If we're not able to do this, then I'd rather just leave the XML glitch in place for now since it's more like an annoyance than an actual problem.

Ideally we can solve this by removing the offending HttpMessageConverter. From previous exploration I suspect this is being created via "new" (i.e. the message converter isn't directly Spring-managed) since there's no Jaxb-based message converter on the application context.

**Comment by eltmon** [comment]
Ok, I determined that some other dependency is pulling in Jax2bRootElementHttpMessageConverter , so I overrode configureMessageConverters to not add it to converters.

I also overrode configureContentNegotiation to only support mediaType json.

File: src/main/java/com/expedia/seiso/MessageConverterConfig.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.List;
+
+@Configuration
+public class MessageConverterConfig extends WebMvcAutoConfiguration.WebMvcAutoConfigurationAdapter {
+
+    @Override
+    public void configureMessageConverters(List<HttpMessageConverter<?>> converters) {
+        // It appears some other dependency is adding Jax2bRootElementHttpMessageConverter, this
+        // will override that behavior.
+        List<HttpMessageConverter<?>> baseConverters = new ArrayList<HttpMessageConverter<?>>();
```
Comment (williewheeler) [inline src/main/java/com/expedia/seiso/MessageConverterConfig.java]: Use diamond notation please, just for consistency with the rest of the code.

List<HttpMessageConverter<?>> baseConverters = new ArrayList<>();

File: src/main/java/com/expedia/seiso/MessageConverterConfig.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            }
+        }
+    }
+    @Override
+    public void configureContentNegotiation(ContentNegotiationConfigurer configurer) {
+        configurer.favorPathExtension(true).
+                favorParameter(false).
+                ignoreAcceptHeader(false).
+                defaultContentType(MediaType.APPLICATION_JSON).
+                mediaType("json", MediaType.APPLICATION_JSON);
```
Comment (williewheeler) [inline src/main/java/com/expedia/seiso/MessageConverterConfig.java]: "json" is not a standard media type and I don't want to add support for nonstandard media types unless there's some good reason to do so.

File: src/main/java/com/expedia/seiso/MessageConverterConfig.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        for(HttpMessageConverter<?> c : baseConverters){
+            if(!(c instanceof Jaxb2RootElementHttpMessageConverter)){
+                converters.add(c);
+            }
+        }
+    }
+    @Override
+    public void configureContentNegotiation(ContentNegotiationConfigurer configurer) {
+        configurer.favorPathExtension(true).
```
Comment (williewheeler) [inline src/main/java/com/expedia/seiso/MessageConverterConfig.java]: Please use . at the start, not at the end, for consistency with the rest of the codebase.

**Pull request merged**